### ISSUE TITLE
Mismatching region results in empty IamDataFrame after processing

### ISSUE THREAD

**Issue by danielhuppmann** [IAMconsortium/nomenclature#70]
The new implementation of the `RegionProcessor.apply()` does the following to a submitted scenario (if a model-region-mapping exists):

 - keep list of model-native regions (possibly with renaming)
 - perform aggregation of common-regions from model-native regions

and returns the combination of these two. This means that any regions in the scenario data that are not defined as model-native-regions will be dropped as part of the renaming.

This means that if a scenario is submitted without any "native regions", the region-processing returns an empty IamDataFrame, which results in an `IndexError: index 0 is out of bounds for axis 0 with size 0` (see [this example](https://data.ece.iiasa.ac.at/ecemf-internal/#/uploads/jobs/346/log)), not yet sure why.

Question: how should this be treated as part of a scenario upload via the Scenario Explorer?
1. `RegionProcessor.apply()` raises a ValueError if the processed scenario is empty
2. the `ixmp-server-workflow` module raises an error if all scenario data is "lost" as part of the project-specific workflow
3. this is actually fine and a user should see a green status (because the workflow performed as expected: no native-regions, no data)

Personally, I think 3 is wrong.

@peterkolp @phackstock, what are your thoughts?

**Comment by phackstock** [comment]
@danielhuppmann agree with you assessment that 3 is not the way to go. It might be technically correct in the sense of 'you submitted nothing, so you get nothing' but it definitely breaks the principle of least surprise. 
Is there a setting imaginable where someone would want to submit an empty scenario?
If this is not the case I would say that `RegionProcessor` should raise an error directly.
The reason why I would prefer this over `ixmp-server-workflow` is that maybe someone goes to the length of testing their data in a juptyer notebook and if we `RegionProcessor` plays along just fine with a empty scenario while `ixmp-server-workflow` complains we again produce surprising results. 
In short, if it should fail it should fail everywhere in the same way.
